Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -86,21 +86,30 @@
         , m_hasWindowProxyPropertyAccessSelector([m_delegate.get() respondsToSelector:@selector(websiteDataStore:domain:didOpenDomainViaWindowOpen:withProperty:directly:)])
         , m_hasDidAllowPrivateTokenUsageByThirdPartyForTestingSelector([m_delegate.get() respondsToSelector:@selector(websiteDataStore:didAllowPrivateTokenUsageByThirdPartyForTesting:forResourceURL:)])
         , m_hasDidExceedMemoryFootprintThresholdSelector([m_delegate.get() respondsToSelector:@selector(websiteDataStore:domain:didExceedMemoryFootprintThreshold:withPageCount:processLifetime:inForeground:wasPrivateRelayed:canSuspend:)])
-        , m_hasWebCryptoMasterKeySelector([m_delegate.get() respondsToSelector:@selector(webCryptoMasterKey)])
+        , m_hasWebCryptoMasterKeySelector([m_delegate.get() respondsToSelector:@selector(getWebCryptoMasterKey:)])
     {
     }
 
 private:
 
-    std::optional<Vector<uint8_t>> webCryptoMasterKey() final
+    void webCryptoMasterKey(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler) final
```

Comment:
This change will make existing `webCryptoMasterKey` unused. 
For consistency, we should either drop the delegate method `webCryptoMasterKey`, or add a new `m_hasWebCryptoMasterKeyAsyncSelector` and make `void webCryptoMasterKey(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler) ` check both delegate methods (otherwise exiting clients might be broken).

Reply:
It would maximize compatibility to call the old selector if present and call the CompletionHandler with the result synchronously then return, and then try calling the new selector if present.

Reply:
webCryptoMasterKey is still used by WebKitLegacy so I have left it there.

Reply:
I think I spoke to soon and totally misunderstood your suggestion. I am gonna keep both as you suggested @szewai  and try the async one first and then the sync one for existing clients.